phys2denoise.py

```diff
@@ -0,0 +1,234 @@
+     path='phys2denoise',
+     description='Creation of regressors for physiological denoising',
+     version=__version__,
+     cite_module=True)
+def phys2denoise(filename, metric_args, outdir='.',
+                 metrics=[crf, rpv, rv, rvt, rrf, 'retroicor_card', 'retroicor_resp'],
+                 debug=False, quiet=False):
+    """
+    Run main workflow of phys2denoise.
+
+    Runs the parser, does some checks on input, then computes the required metrics.
+
+    Notes
+    -----
+    The code was greatly copied from phys2bids (copyright the physiopy community)
+
+    """
+    # Check options to make them internally coherent pt. I
+    # #!# This can probably be done while parsing?
+    outdir = os.path.abspath(outdir)
+    os.makedirs(outdir)
+    os.makedirs(os.path.join(outdir, 'code'))
+    log_path = os.path.join(outdir, 'code', 'logs')
+    os.makedirs(log_path)
+
+    # Create logfile name
+    basename = 'phys2denoise_'
+    extension = 'tsv'
+    isotime = datetime.datetime.now().strftime('%Y-%m-%dT%H%M%S')
+    logname = os.path.join(log_path, (basename + isotime + '.' + extension))
+
+    # Set logging format
+    log_formatter = logging.Formatter(
+        '%(asctime)s\t%(name)-12s\t%(levelname)-8s\t%(message)s',
+        datefmt='%Y-%m-%dT%H:%M:%S')
+
+    # Set up logging file and open it for writing
+    log_handler = logging.FileHandler(logname)
+    log_handler.setFormatter(log_formatter)
+    sh = logging.StreamHandler()
+
+    if quiet:
+        logging.basicConfig(level=logging.WARNING,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+    elif debug:
+        logging.basicConfig(level=logging.DEBUG,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+    else:
+        logging.basicConfig(level=logging.INFO,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+
+    version_number = _version.get_versions()['version']
+    LGR.info(f'Currently running phys2denoise version {version_number}')
+    LGR.info(f'Input file is {filename}')
+
+    # Save call.sh
+    arg_str = ' '.join(sys.argv[1:])
+    call_str = f'phys2denoise {arg_str}'
+    f = open(os.path.join(log_path, 'call.sh'), "a")
+    f.write(f'#!bin/bash \n{call_str}')
+    f.close()
+
+    # Check options to make them internally coherent pt. II
+    # #!# This can probably be done while parsing?
+    # filename, ftype = utils.check_input_type(filename)
+
+    if not os.path.isfile(filename) and filename is not None:
+        raise FileNotFoundError(f'The file {filename} does not exist!')
+
+    # Read input file
+    physio = np.genfromtxt(filename)
+
+    # Prepare pandas dataset
+    regr = pd.DataFrame()
+
+    # Goes through the list of metrics and calls them
+    for metric in metrics:
+        if metrics == 'retroicor_card':
```

Comment:
Ah and, shouldn this be ``metric`` instead of ``metrics``, since you are inside the for loop, I think you should change the name of this ``var`` so there is no confusion